modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/LogStorageBudgetsModule.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.raft.jraft.core;
+
+import java.util.Map;
+import java.util.function.Function;
+import org.apache.ignite.configuration.schemas.table.LogStorageBudgetView;
+import org.apache.ignite.raft.jraft.storage.impl.LogStorageBudget;
+
+/**
+ * Used to add {@link LogStorageBudget} factories using {@link java.util.ServiceLoader}.
+ */
+public interface LogStorageBudgetsModule {
+    /**
+     * Returns mapping from budget names to budget factories for the budgets supported by this module.
+     *
+     * @return Mapping from budget names to budget factories for the budgets supported by this module.
+     */
+    Map<String, Function<? super LogStorageBudgetView, LogStorageBudget>> budgetFactories();
```

Comment:
Why not create a separate factory interface? it is also not necessary to take the form, you can just config.

Reply:
Created an interface, thanks for the suggestion. But it looks like view is enough here: we only need to read the config and don't need the flexibility given by the full `Config`.